### ISSUE TITLE
Got 0 tweets

### ISSUE THREAD

**Issue by danmlf1** [taspinar/twitterscraper#344]

**Comment by miatang13** [comment]
I have the same issue. "twitterscraper Trump --limit 1000 --output=tweets.json" gives an empty json file.

**Comment by calebgrisell** [comment]
same issue :(

**Comment by kaushalvivek** [comment]
Same issue, getting 0 tweets.

**Comment by Aivi001** [comment]
Yes, I'm having the same problem, getting the following error message when "getting 0 tweets"

 raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPSConnectionPool(host='twitter.com', port=443): Max retries exceeded with url: /search?f=tweets&vertical=default&q=%23Harami%20since%3A2020-08-22%20until%3A2020-08-24&l=en (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f5a3decb510>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution'))

**Comment by GivenToFlyCoder** [comment]
I have the same issue, 0 tweets using query_tweets...

**Comment by eugenelee123** [comment]
It looks like there is a pull request that addresses this issue. Here is the [link](https://github.com/taspinar/twitterscraper/pull/337). I cloned the repo, and just edited `query.py` with what @sm-ghavami changed on the pull request and it seems to work just fine. There's likely a better way to do this though.

**Comment by jason1991220** [comment]
It doesn't work and return 0 tweet again. please help.

**Comment by christiangfv** [comment]
ooh bad, doesn't work function query_tweets from_user() nor query_tweets(), since yesterday. please help. 
maybe here is information https://blog.twitter.com/developer/en_us/topics/tips/2020/understanding-the-new-tweet-payload.html

**Comment by gautampal1947** [comment]
0 Tweets again. Please help.

**Comment by zhicheng0501** [comment]
> ooh bad, doesn't work function query_tweets from_user() nor query_tweets(), since yesterday. please help. 
> maybe here is information https://blog.twitter.com/developer/en_us/topics/tips/2020/understanding-the-new-tweet-payload.html

The same error.
I agree with you.

**Comment by tpsvld** [comment]
Easy Twitter enabled Javascript and shut down the backdor this code used 
In detail:
Twitter has been developing since beginning of 2020 a new HTML Frontend with Randomly generated HTML Description Tags.
The scoll mechanism to load new tweets was already changed to work with javascript and unique session ID's in january 2020.
The URL "https://twitter.com/i/search/ with the &max_position=cursor" has been disabled in twitter.

It was surprising that with the russia collusion and other horse shit they didnt start to work to protect the data. 

So as i said in April, the software is probably dead.

**Comment by tpsvld** [comment]
> ooh bad, doesn't work function query_tweets from_user() nor query_tweets(), since yesterday. please help.
> maybe here is information https://blog.twitter.com/developer/en_us/topics/tips/2020/understanding-the-new-tweet-payload.html

This doesnt work with the Twitter API not even close.

They changed the HTML Page to avoid non conscented web scrapping.

**Comment by AveryData** [comment]
So is this library just donzo? It's over?

**Comment by zhicheng0501** [comment]
> So is this library just donzo? It's over? 

Temporarily. Yeah.
Hope it will be back soon

**Comment by pavstar619** [comment]
Thought I'd use this for my project. Guess that's not gonna happen anytime soon :(

**Comment by giulionf** [comment]
I think this man found a solution: https://github.com/Foo-Manroot/tweet-feed/issues/1

**Comment by edmangog** [comment]
Same here. Trying the branch of selenium alternatively.

**Comment by sudz4** [comment]
Is there another good similar library that people are using?

**Comment by adamzvx** [comment]
same to me

**Comment by Simon-Smith-NKU** [comment]
same issue ，twitter disgust,

**Comment by qarampage** [comment]
Even I got 0 Tweets count. I posted a new tweet and still got Zero tweets.  I spent a lot of time understanding the workflow and got 0 !

**Comment by sudz4** [comment]
just use tweepy instead

import tweepy as tw
#Importing twitter access keys from a separate file in this project folder:
from credentials import * #Using keys as a variable

#Setting up Twitter API connection:
def twitter_creds():
    """
    Connects to Twitter API.
    For confidentiality purposes the access keys are in a separate file in the project folder.
    Credential file is saved as credentials.py.

    
    """

    #Authenticate credentials using keys:
    auth = tw.OAuthHandler(CONSUMER_KEY, CONSUMER_SECRET) #tells twitter I am valid user
    auth.set_access_token(ACCESS_TOKEN, ACCESS_SECRET)

    #Connect with API and authenticate:
    api = tw.API(auth) #speaks to twitter
    return api



#Finding some AI Tweets

search_words = "artificial+intelligence"
date_since = "2020-1-1"
#2020-3-12 outputted 669 tweets
#For fun let's see if we can get a little closer to 1000
#2020-3-12 gives me 673 tweets. 4 tweets in 3 months??? I don't know...
#2020-1-1 gets us to the limit set below of 1000
tweets = tw.Cursor(api.search,
              q=search_words,
              lang="en",
              since=date_since).items(1000)

for tweet in tweets:
print(tweet.text)

**Comment by do-me** [comment]
As twitterscraper is (temporarily) dead, I thought I'd share my favorite alternative scraper. @mattwsutherland your solution requires an API-key (which is super hard to get) so unfortunately it's useless to most people here. 

Simply go for [TweetScraper](https://github.com/jonbakerfish/TweetScraper) - for me its working perfectly. 
Also see this [medium article ](https://medium.com/@kevin.a.crystal/scraping-twitter-with-tweetscraper-and-python-ea783b40443b) providing a useful pandas snippet to set up a nicely formatted dataframe. 

Installation for Ubuntu is straightforward as described. For Windows just do the following: 
```
git clone https://github.com/jonbakerfish/TweetScraper.git
conda create -n tweetscraper python=3.7.7 -y
conda activate tweetscraper
conda install -y -c conda-forge scrapy ipython ipdb
pip install scrapy-selenium
```
Change to directory "TweetScraper" and test with 
```
scrapy crawl TweetScraper -a query="foo,#bar"
```

**Comment by qarampage** [comment]
Thanks I will definitely 👍 try that

**Comment by ghost** [comment]
still, the issue is continuing, Anyone finds a solution or another work around?

**Comment by Altimis** [comment]
@lavarthan 
@qarampage 

All twitter scrapers dont seem to work anymore. I tried to use selenium to simply scrap maximum tweets between two chosen dates for given quiries. Check my work on : [Scweet](https://github.com/Altimis/Scweet). let me know if you need any clarification.

**Comment by Guolin1996** [comment]
I think this scrapper is dead and may not be able untill somebody make changes.

**Comment by Altimis** [comment]
@Guolin1996  Yes unfortunately ...

**Comment by amitchandnia** [comment]
> I have the same issue. "twitterscraper Trump --limit 1000 --output=tweets.json" gives an empty json file.

where to set Author API in this code?

**Comment by shawnngtq** [comment]
Commenting to watch, same issue here

**Comment by woaying** [comment]
I guess it's still dead. Just tried, got 0 tweets.

**Comment by Hat52** [comment]
Still 0 tweets

**Comment by raygorous** [comment]
sad. still 0 tweets. is this related to Elon Musk?
